src/bucket/BucketManagerImpl.cpp

```diff
@@ -1521,4 +1523,72 @@ BucketManagerImpl::getConfig() const
 {
     return mApp.getConfig();
 }
+
+std::string
+to_string(LedgerEntryTypeAndDurability const type)
+{
+    switch (type)
+    {
+    case LedgerEntryTypeAndDurability::ACCOUNT:
+        return "ACCOUNT";
+    case LedgerEntryTypeAndDurability::TRUSTLINE:
+        return "TRUSTLINE";
+    case LedgerEntryTypeAndDurability::OFFER:
+        return "OFFER";
+    case LedgerEntryTypeAndDurability::DATA:
+        return "DATA";
+    case LedgerEntryTypeAndDurability::CLAIMABLE_BALANCE:
+        return "CLAIMABLE_BALANCE";
+    case LedgerEntryTypeAndDurability::LIQUIDITY_POOL:
+        return "LIQUIDITY_POOL";
+    case LedgerEntryTypeAndDurability::TEMPORARY_CONTRACT_DATA:
+        return "TEMPORARY_CONTRACT_DATA";
+    case LedgerEntryTypeAndDurability::PERSISTENT_CONTRACT_DATA:
+        return "PERSISTENT_CONTRACT_DATA";
+    case LedgerEntryTypeAndDurability::CONTRACT_CODE:
+        return "CONTRACT_CODE";
+    case LedgerEntryTypeAndDurability::CONFIG_SETTING:
+        return "CONFIG_SETTING";
+    case LedgerEntryTypeAndDurability::TTL:
+        return "TTL";
+    default:
+        throw std::runtime_error(
+            fmt::format(FMT_STRING("unknown LedgerEntryTypeAndDurability {:d}"),
+                        static_cast<uint32_t>(type)));
+    }
+}
+
+void
+BucketManagerImpl::reportBucketEntryCountMetrics()
+{
+    auto bucketEntryCounters = mBucketList->sumBucketEntryCounters();
+    for (auto& [type, count] : bucketEntryCounters.mEntryTypeCounts)
+    {
+        auto countsCounter = mBucketListEntryCounts.find(type);
+        if (countsCounter == mBucketListEntryCounts.end())
```

Comment:
This lazy construction is a little difficult to keep up with all the iterators and pairs floating around. I think it might be simpler to just construct all the metrics in the BucketManager constructor instead of having lazy instantiation logic here.